manager/state/raft/storage.go

```diff
@@ -215,26 +219,23 @@ func (n *Node) doSnapshot(ctx context.Context, raftConfig api.RaftConfig) {
 	<-viewStarted
 }
 
-func (n *Node) restoreFromSnapshot(data []byte, forceNewCluster bool) error {
+func (n *Node) restoreFromSnapshot(data []byte, forceNewCluster bool) ([]*api.RaftMember, error) {
 	var snapshot api.Snapshot
 	if err := snapshot.Unmarshal(data); err != nil {
-		return err
+		return nil, err
 	}
 	if snapshot.Version != api.Snapshot_V0 {
-		return fmt.Errorf("unrecognized snapshot version %d", snapshot.Version)
+		return nil, fmt.Errorf("unrecognized snapshot version %d", snapshot.Version)
 	}
 
 	if err := n.memoryStore.Restore(&snapshot.Store); err != nil {
-		return err
+		return nil, err
 	}
 
 	oldMembers := n.cluster.Members()
```

Comment:
(as an addendum, should the removed list be propagated to the new cluster, if `forceNewCluster` is true?)